src/native/managed/cdacreader/tests/CodeVersionsTests.cs

```diff
@@ -63,202 +64,80 @@ public static MockMethodDesc CreateVersionable (TargetPointer selfAddress, Targe
             };
         }
     }
+
     internal class MockCodeBlockStart
     {
         public TargetCodePointer StartAddress { get; set;}
         public uint Length { get; set; }
-
-        public bool Contains(TargetPointer ip) =>  ip >= StartAddress && ip < StartAddress + Length;
-        public bool Contains(TargetCodePointer ip) => Contains(ip.AsTargetPointer);
         public MockMethodDesc MethodDesc {get; set;}
-        public TargetPointer MethodDescAddress => MethodDesc.Address;
     }
 
-    internal class MockExecutionManager : IExecutionManager
-    {
-        private IReadOnlyCollection<MockCodeBlockStart> _codeBlocks;
+    private Mock<IRuntimeTypeSystem> mockRuntimeTypeSystem = new(MockBehavior.Strict);
+    private Mock<IExecutionManager> mockExecutionManager = new(MockBehavior.Strict);
+    private Mock<ILoader> mockLoader = new(MockBehavior.Strict);
 
-        public MockExecutionManager(IReadOnlyCollection<MockCodeBlockStart> codeBlocks)
-        {
-            _codeBlocks = codeBlocks;
-        }
-
-        private MockCodeBlockStart? FindCodeBlock(TargetPointer ip)
-        {
-            if (ip == TargetPointer.Null)
-            {
-                return null;
-            }
-            foreach (var block in _codeBlocks)
-            {
-                if (block.Contains(ip))
-                {
-                    return block;
-                }
-            }
-            return null;
-        }
-
-        CodeBlockHandle? IExecutionManager.GetCodeBlockHandle(TargetCodePointer ip)
-        {
-            var block = FindCodeBlock(ip.AsTargetPointer);
-            if (block == null)
-                return null;
-            return new CodeBlockHandle(ip.AsTargetPointer);
-        }
-
-        TargetCodePointer IExecutionManager.GetStartAddress(CodeBlockHandle codeInfoHandle) => FindCodeBlock(codeInfoHandle.Address)?.StartAddress ?? TargetCodePointer.Null;
-        TargetPointer IExecutionManager.GetMethodDesc(CodeBlockHandle codeInfoHandle) => FindCodeBlock(codeInfoHandle.Address)?.MethodDescAddress ?? TargetPointer.Null;
+    private void AddCodeBlock(MockCodeBlockStart block)
```

Comment:
Do you think `MockCodeBlockStart` and friends are still helpful? Versus having these method just take in the address, length, method desc, etc. as separate parameters.

Reply:
I think they are still helpful to encapsulate state related to a mock. I think if we removed them, it would make tests with multiple MethodDescs confusing due to keeping track of multiple similar fields.